lib/nodeUtil.js

```diff
@@ -0,0 +1,101 @@
+/*!
+ * Copyright (c) 2021, Salesforce.com, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of Salesforce.com nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+const customInspectSymbol = Symbol.for("nodejs.util.inspect.custom");
+
+function inspectFallback(val) {
```

Comment:
This implementation is specific to `MemoryCookieStore`. I don't think that it should live in the generic `util.inspect`.

Also, if we implement the custom inspect in the class, then all we need `util` for is `util.inspect.custom`. That symbol was added to the global registry (`Symbol.for("nodejs.util.inspect.custom")`) in node [v10.12.0](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V10.md#10.12.0), so once we drop support for earlier versions (our next major release, I hope), we can use that and completely remove the need for `util`.

In the meantime, if all we need is `util.inspect.custom`, there's nothing to implement, so we don't really need a separate file. We can just wrap `require("util")` in a try/catch in the relevant files, and change `if (util.inspect.custom)` to `if (util)`. Testing that is a bit more complex, though, as we'd need to hook into `require`.

Reply:
yes, it is specific to `MemoryCookieStore` which, i agree, is a bit confusing but, internally it is consistent.   when running in a node environment the existing output is unchanged and when running in a native environment there is no `util.inspect` to use but anyone directly calling the `inspect` function attached to those objects will receive the same output as in node.  but i'll inline that implementation into `MemoryCookieStore` since that will be less confusing 😅

Reply:
@wjhsf moved some stuff around so the internal usage of `util` is split into 2 concerns:
* retrieving the custom inspect symbol so a custom inspect function can be attached to `Cookie` and `MemoryCookieStore`
* providing an equivalent inspect output for `MemoryCookieStore` when explicitly calls `util.inspect(...)`

i'd like to avoid hooking into `require` to make this testable.  instead, the `utilHelper.getCustomInspectSymbol` and `utilHelper.getUtilInspect` functions have hooks in place to simulate fallbacks.  

the test suite has also been expanded to run through all the fallback scenarios to verify it matches the existing behavior.